src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -21,6 +22,21 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
         private static readonly IReadOnlyList<string> HtmlTriggerCharacters = new[] { "<", "&", "\\", "/", "'", "\"", "=", ":", " " };
         private static readonly IReadOnlyList<string> AllTriggerCharacters = CSharpTriggerCharacters.Concat(HtmlTriggerCharacters).ToArray();
 
+        private static readonly IReadOnlyCollection<string> _keywords = new string[] {
+            "for", "foreach", "while", "switch", "lock",
+            "case", "if", "try", "do", "using"
+        };
+
+        private static readonly IEnumerable<CompletionItem> _keywordCompletionItems = _keywords.Select(k => new CompletionItem
```

Comment:
With the change to `IEnumerable` it means every time you iterate this collection it's going to re-evaluate / recompute all of the completion items. Is that what you intended?